### PR TITLE
Set Newsletter Config

### PR DESCRIPTION
**Pull request by sedonik**
### Description (*)

After created Magento\Newsletter\Model\Config it needs to update Magento\Newsletter\Observer\PredispatchNewsletterObserver with new implementation.

### Manual testing scenarios (*)

Create Customer from Admin Panel.

### Questions or comments

Related to [PR](https://github.com/magento/graphql-ce/pull/763).

### CODE REVIEW

File: app/code/Magento/Newsletter/Observer/PredispatchNewsletterObserver.php [hunk 1]
```diff
@@ -38,11 +39,16 @@ class PredispatchNewsletterObserver implements ObserverInterface
...
      * @param ScopeConfigInterface $scopeConfig
      * @param UrlInterface $url
+     * @param Config $config
      */
-    public function __construct(ScopeConfigInterface $scopeConfig, UrlInterface $url)
-    {
+    public function __construct(
+        ScopeConfigInterface $scopeConfig,
+        UrlInterface $url,
+        Config $config
```
Comment (lenaorobei) [inline app/code/Magento/Newsletter/Observer/PredispatchNewsletterObserver.php]: Please follow BC guide here. 
https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/
`Adding a constructor parameter` section.

**Comment by lenaorobei** [comment]
Please combine those PRs in a way that GraphQl related code is in one PR, core - in the second one. 
For example, you need to move `app/code/Magento/Newsletter/Model/Config.php` here.

**Comment by sedonik** [comment]
@lenaorobei,
I've updated the branch.
Pls, check.

File: app/code/Magento/Newsletter/Model/Config.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    ) {
+        $this->scopeConfig = $scopeConfig;
+    }
+
+    /**
+     * Returns newsletter's enabled status
+     *
+     * @return bool
+     */
+    public function isActive($scopeType = ScopeConfigInterface::SCOPE_TYPE_DEFAULT)
```
Comment (lenaorobei) [inline app/code/Magento/Newsletter/Model/Config.php]: `Missing @param for argument in method annotation`

Comment (lenaorobei) [inline app/code/Magento/Newsletter/Model/Config.php]:
```suggestion
    public function isActive($scopeType = ScopeConfigInterface::SCOPE_TYPE_DEFAULT): bool
```

File: app/code/Magento/Newsletter/Model/Config.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Newsletter configuration model
+ */
+class Config
+{
+    /**
+     * Configuration path to newsletter active setting
+     */
+    const XML_PATH_NEWSLETTER_ACTIVE = 'newsletter/general/active';
```
Comment (lenaorobei) [inline app/code/Magento/Newsletter/Model/Config.php]:
```suggestion
    private const XML_PATH_NEWSLETTER_ACTIVE = 'newsletter/general/active';
```

**Comment by sedonik** [comment]
@lenaorobei ,

Thanks for the update.
I've added commit with suggestions.
Pls, check one more time.

**Comment by lenaorobei** [comment]
@magento run Functional Tests build Integration Tests build

**Comment by lenaorobei** [comment]
@magento run Functional Tests build, Integration Tests build

**Comment by lenaorobei** [comment]
@sedonik tests fail with `Undefined class constant 'XML_PATH_NEWSLETTER_ACTIVE'` message. To keep BC we cannot remove constants but need to mark it as deprecated. All usages of `PredispatchNewsletterObserver::XML_PATH_NEWSLETTER_ACTIVE` need to be replace with `Config ::XML_PATH_NEWSLETTER_ACTIVE`.

File: app/code/Magento/Newsletter/Observer/PredispatchNewsletterObserver.php [hunk 4]
```diff
@@ -12,16 +12,18 @@
...
 /**
  * Class PredispatchNewsletterObserver
  */
 class PredispatchNewsletterObserver implements ObserverInterface
 {
     /**
-     * Configuration path to newsletter active setting
+     * @var Newsletter Config
      */
-    const XML_PATH_NEWSLETTER_ACTIVE = 'newsletter/general/active';
```
Comment (lenaorobei) [inline app/code/Magento/Newsletter/Observer/PredispatchNewsletterObserver.php]: Please mark as deprecated and add `use Magento\Newsletter\Model\Config::XML_PATH_NEWSLETTER_ACTIVE instead` comment.

**Comment by sedonik** [comment]
@lenaorobei,
I've updated.

**Comment by lenaorobei** [comment]
@sedonik we also need to check usages of deprecated constant and replace with newly introduced one.

**Comment by sedonik** [comment]
@lenaorobei
Updated.
Please check.

**Comment by lenaorobei** [comment]
@sedonik please see my fixes I pushed to your branch. Once we deliver this PR to `2.3-develop` you will need to pull latest changes to https://github.com/magento/graphql-ce/pull/763 and finish this PR.

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-5636](https://jira.corp.magento.com/browse/ENGCOM-5636) has been created to process this Pull Request
:eight_spoked_asterisk: @lenaorobei, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by sedonik** [comment]
Ok, @lenaorobei,
I saw the changes. Thank you.

**Pull request merged**